plugins/processors/k8sdecorator/stores/utils_test.go

```diff
@@ -4,17 +4,99 @@
 package stores
 
 import (
-	"github.com/docker/docker/pkg/testutil/assert"
+	"strconv"
 	"testing"
+	"time"
+
+	"github.com/docker/docker/pkg/testutil/assert"
 )
 
 func TestUtils_parseDeploymentFromReplicaSet(t *testing.T) {
-	assert.Equal(t, "", parseDeploymentFromReplicaSet("cloudwatch-agent"))
-	assert.Equal(t, "cloudwatch-agent", parseDeploymentFromReplicaSet("cloudwatch-agent-42kcz"))
+	testcases := []struct {
+		name        string
+		inputString string
+		expected    string
+	}{
+		{
+			name:        "Get ReplicaSet Name with unallowed characters",
```

Comment:
One additional test for allowed characters but less than 3

Reply:
Reasonable. Added it and it passed. Currently I am working on integration test with eks to ensure the naming convention.